src/wp-includes/class-wp-scripts.php

```diff
@@ -512,31 +476,124 @@ public function add_inline_script( $handle, $data, $position = 'after', $standal
+		if ( $display ) {
+			$output = $this->get_inline_script_tag( $handle, $position );
+			echo $output;
+		} else {
+			$output = $this->get_inline_script_data( $handle, $position );
+		}
 		if ( empty( $output ) ) {
 			return false;
 		}
+		return $output;
+	}
 
-		$output = trim( implode( "\n", $output ), "\n" );
+	/**
+	 * Gets data for inline scripts registered for a specific handle.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle   Name of the script to add the inline script to.
+	 *                         Must be lowercase.
+	 * @param string $position Optional. Whether to add the inline script
+	 *                         before the handle or after. Default 'after'.
+	 * @return string Inline script, which may be empty string.
+	 */
+	public function get_inline_script_data( $handle, $position = 'after' ) {
+		$data = $this->get_data( $handle, $position );
+		if ( empty( $data ) || ! is_array( $data ) ) {
+			return '';
+		}
 
-		if ( $display ) {
-			if ( 'after-non-standalone' === $position ) {
-				$script_output = "<script%1\$s id='%2\$s-js-after' data-wp-executes-after='%2\$s'>\n%4\$s\n</script>\n";
-			} else {
-				$script_output = "<script%1\$s id='%2\$s-js-%3\$s'>\n%4\$s\n</script>\n";
-			}
+		return trim( implode( "\n", $data ), "\n" );
+	}
+
+	/**
+	 * Gets tags for inline scripts registered for a specific handle.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle   Name of the script to add the inline script to.
+	 *                         Must be lowercase.
+	 * @param string $position Optional. Whether to add the inline script
+	 *                         before the handle or after. Default 'after'.
+	 * @return string Inline script, which may be empty string.
+	 */
+	public function get_inline_script_tag( $handle, $position = 'after' ) {
+		$js = $this->get_inline_script_data( $handle, $position );
+		if ( empty( $js ) ) {
+			return '';
+		}
 
-			printf(
-				$script_output,
-				$this->type_attr,
-				esc_attr( $handle ),
-				esc_attr( $position ),
-				$output
+		$id   = "{$handle}-js-{$position}";
+		$deps = $this->registered[ $handle ]->deps;
+
+		if ( $this->should_delay_inline_script( $handle, $position ) ) {
+			$attributes = array(
+				'id'   => $id,
+				'type' => 'text/plain',
 			);
+			if ( $deps ) {
+				$attributes['data-wp-deps'] = implode( ',', $deps );
+			}
+			return wp_get_inline_script_tag( $js, $attributes );
+		} else {
+			return wp_get_inline_script_tag( $js, compact( 'id' ) );
 		}
+	}
 
-		return $output;
+	/**
+	 * Determines whether an inline script should be delayed.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle   Name of the script to add the inline script to.
+	 *                         Must be lowercase.
+	 * @param string $position Optional. Whether to add the inline script
+	 *                         before the handle or after. Default 'after'.
```

Comment:
```suggestion
	 * @param string $position Whether to check inline scripts
	 *                         in the before or after position.
```

Or similar. Removed default as well, as there is none, as well as optional, as it is not optional given the lack of a default value.